src/main/java/com/lafaspot/ja3_4java/JA3Signature.java

```diff
@@ -0,0 +1,292 @@
+package com.lafaspot.ja3_4java;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+
+/**
+ * Generates JA3 signature based on the implementation described at https://github.com/salesforce/ja3.
+ *
+ */
+public final class JA3Signature {
+    /**
+     * Handshake byte.
+     */
+    private static final byte HANDSHAKE = 22;
+    /**
+     * Client hello.
+     */
+    private static final byte CLIENT_HELLO = 1;
+    /**
+     * Client hello random lenght.
+     */
+    private static final byte CLIENT_HELLO_RANDOM_LEN = 32;
+
+    /**
+     * Values to account for GREASE (Generate Random Extensions And Sustain Extensibility) as described here:
+     * https://tools.ietf.org/html/draft-davidben-tls-grease-01.
+     */
+    private static final int[] GREASE = new int[] { 0x0a0a, 0x1a1a, 0x2a2a, 0x3a3a, 0x4a4a, 0x5a5a, 0x6a6a, 0x7a7a, 0x8a8a, 0x9a9a, 0xaaaa, 0xbaba,
+            0xcaca, 0xdada, 0xeaea, 0xfafa };
+
+
+    /**
+     * Private constructor.
+     */
+    private JA3Signature() {
+        // prevent instantiation
+    }
+
+    /**
+     * Calculate JA3 string from a ClientHello packet.Note that we do not compute an MD5 hash here.
+     *
+     * @param packet packet to inspect
+     * @return JA3 fingerprint or null if no TLS ClientHello detected in given packet
+     * @see <a href="https://github.com/salesforce/ja3">Original JA3 implementation</a>
+     */
+    public static String ja3Signature(final ByteBuffer packet) {
```

Comment:
Why static? we could have an instance in our app context (imap/pop) and pass it in where needed. Talk to vikram also.

Reply:
spoke to Vikram. Wont be super useful adding this to the app context

Reply:
removed static